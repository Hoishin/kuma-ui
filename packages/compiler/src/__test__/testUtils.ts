import { compile } from "..";

type Compiled = ReturnType<typeof compile>;

export function getExpectSnapshot(result: Compiled) {
  return `
  ${result.css}
  
  ${result.code}
  `;
}
