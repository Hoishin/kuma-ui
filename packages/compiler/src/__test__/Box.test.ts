import { describe, test, expect } from "vitest";
import { compile } from "../compile";
import { getExpectSnapshot } from "./testUtils";

describe("", () => {
  test("", () => {
    // Arrange
    const code = `
        import { Box } from '@kuma-ui/core'

        const App = () => {
            return <Box color='red'>hello world</Box>
        }
        `;
    const result = compile(code, "test", { Box: "Box" });
    expect(getExpectSnapshot(result)).toMatchSnapshot();
  });
});
