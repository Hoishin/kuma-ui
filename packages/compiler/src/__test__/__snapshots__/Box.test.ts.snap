// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[` 1`] = `
"
  
  
  
        import { Box } from '@kuma-ui/core'

        const App = () => {
            return <Box color='red'>hello world</Box>
        }
        
  "
`;
